reference/definitions.json

```diff
@@ -0,0 +1,396 @@
+        "type": "String[]",
+        "desc": "The interface's IPv4 addresses"
+      },
+      {
+        "name": "ipv6",
+        "type": "String[]",
+        "desc": "The interface's IPv6 addresses"
+      },
+      {
+        "name": "broadcast",
+        "type": "String",
+        "desc": null
+      },
+      {
+        "name": "netmask",
+        "type": "String",
+        "desc": "The interface's subnet mask"
+      },
+      {
+        "name": "read_bytes",
+        "type": "Long",
+        "desc": "The number of bytes read from the interface"
+      },
+      {
+        "name": "write_bytes",
+        "type": "Long",
+        "desc": "The number of bytes written to the interface"
+      },
+      {
+        "name": "read_packets",
+        "type": "Long",
+        "desc": "The number of packets read from the interface"
+      },
+      {
+        "name": "write_packets",
+        "type": "Long",
+        "desc": "The number of packets written to the interface"
+      },
+      {
+        "name": "read_errors",
+        "type": "Long",
+        "desc": "The number of read errors"
+      },
+      {
+        "name": "write_errors",
+        "type": "Long",
+        "desc": "The number of write errors"
+      },
+      {
+        "name": "read_drops",
+        "type": "Long",
+        "desc": "The number of read drops"
+      },
+      {
+        "name": "write_drops",
+        "type": "Long",
+        "desc": "The number of write drops"
+      },
+      {
+        "name": "write_collisions",
+        "type": "Long",
+        "desc": "The number of write collisions"
+      },
+      {
+        "name": "link_speed",
+        "type": "Long",
+        "desc": "The interface's maximum speed in bytes"
+      },
+      {
+        "name": "default_gateway",
+        "type": "Boolean",
+        "desc": null
+      },
+      {
+        "name": "flag_up",
+        "type": "Boolean",
+        "desc": null
+      },
+      {
+        "name": "flag_running",
+        "type": "Boolean",
+        "desc": null
+      },
+      {
+        "name": "flag_loopback",
+        "type": "Boolean",
+        "desc": null
+      },
+      {
+        "name": "flag_multicast",
+        "type": "Boolean",
+        "desc": null
+      },
+      {
+        "name": "luid",
+        "type": "Long",
+        "desc": "The interface's locally unique identifier",
+        "compatible": [
+          "windows"
+        ]
+      },
+      {
+        "name": "guid",
+        "type": "String",
+        "desc": "The interface's globally unique identifier",
+        "compatible": [
+          "windows"
+        ]
+      },
+      {
+        "name": "paused",
+        "type": "Boolean",
+        "desc": "Whether the interface is in a paused state",
+        "compatible": [
+          "windows"
+        ]
+      },
+      {
+        "name": "lowPower",
+        "type": "Boolean",
+        "desc": "Whether the interface is in a low-power state",
+        "compatible": [
```

Comment:
I have an `extension` field that encodes an attribute's requirement of external add-ons, but I don't do anything with it in the drivers yet.

I removed permissions requirements from the spec because it's an implementation detail and makes more sense to declare it as an annotation on the method that requires permissions.

The problem with the `incompatible` field that we came up with in the initial YAML spec is that it pushes the attribute down into platform-specific territory. I'll add a follow up comment with a possible solution for this that I came up with.

As for the uptime example, you're saying that maybe the attribute should be compatible only on later Windows? I'm not sure how this could be done without arbitrating `WindowsXP` classes for example (which would be a nightmare).

Reply:
It's less a compatibility than a "document the results" thing.  I'm just recalling there are many places in the Javadocs where I'm commenting about what you can and can't get on various OS's.